app/assets/javascripts/discourse/tests/acceptance/user-preferences-account-user-status-test.js

```diff
@@ -0,0 +1,128 @@
+import {
+  acceptance,
+  query,
+  updateCurrentUser,
+} from "discourse/tests/helpers/qunit-helpers";
+import { click, fillIn, visit } from "@ember/test-helpers";
+import { test } from "qunit";
+
+async function openUserStatusModal() {
+  await click(".pref-user-status .btn-default");
+}
+
+async function pickEmoji(emoji) {
+  await click(".btn-emoji");
+  await fillIn(".emoji-picker-content .filter", emoji);
+  await click(".results .emoji");
+}
+
+async function setStatus(status) {
+  await openUserStatusModal();
+  await pickEmoji(status.emoji);
+  await fillIn(".user-status-description", status.description);
+  await click(".modal-footer .btn-primary"); // save and close modal
+}
+
+acceptance("User Profile - Account - User Status", function (needs) {
+  const username = "eviltrout";
+  const status = {
+    emoji: "tooth",
+    description: "off to dentist",
+  };
+
+  needs.user({ username, status });
+
+  test("doesn't render status block if status is disabled in site settings", async function (assert) {
+    this.siteSettings.enable_user_status = false;
+    await visit(`/u/${username}/preferences/account`);
+    assert.notOk(query(".pref-user-status"));
+  });
+
+  test("renders status block if status is enabled in site settings", async function (assert) {
+    this.siteSettings.enable_user_status = true;
+
+    await visit(`/u/${username}/preferences/account`);
+
+    assert.ok(
+      query(".pref-user-status .user-status-message"),
```

Comment:
I would probably use `exists` instead of `query` all the time... but can't say it's very different :p

Reply:
`exists` is better for sure 👍